File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/IssuerStatePublisher.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.Bridge;
import com.bloock.sdk.bridge.proto.AuthenticityEntities;
import com.bloock.sdk.bridge.proto.Config;
import com.bloock.sdk.bridge.proto.IdentityV2;
import com.bloock.sdk.bridge.proto.Shared;

public class IssuerStatePublisher {
  private final String issuerDid;
  private AuthenticityEntities.Signer signer;
  private final Config.ConfigData configData;

  public IssuerStatePublisher(String issuerDid, Config.ConfigData configData) {
    this.issuerDid = issuerDid;
    this.configData = configData;
    this.signer = null;
  }

  public IssuerStatePublisher withSigner(com.bloock.sdk.entity.authenticity.Signer signer) {
    this.signer = signer.toProto();
    return this;
  }

  public IssuerStateReceipt build() throws Exception {
    Bridge bridge = new Bridge();

    IdentityV2.PublishIssuerStateRequest req = IdentityV2.PublishIssuerStateRequest.newBuilder()
        .setConfigData(this.configData)
        .setIssuerDid(this.issuerDid)
        .setSigner(this.signer)
        .build();

    IdentityV2.PublishIssuerStateResponse response = bridge.getIdentityV2().publishIssuerState(req);

    if (response.getError() != Shared.Error.getDefaultInstance()) {
      throw new Exception(response.getError().getMessage());
    }

    return IssuerStateReceipt.fromProto(response.getStateReceipt());
  }
}

File: languages/php/Bloock/Entity/IdentityV2/IssuerStatePublisher.php
<?php

namespace Bloock\Entity\IdentityV2;

use Bloock\Bridge\Bridge;
use Bloock\ConfigData;
use Bloock\PublishIssuerStateRequest;
use Bloock\Signer as SingerProto;
use Bloock\Entity\Authenticity\Signer;
use DateTime;
use Exception;

class IssuerStatePublisher
{
    private string $issuerDid;
    private SingerProto $signer;
    private ConfigData $configData;

    public function __construct(string $issuerDid, ConfigData $configData)
    {
        $this->issuerDid = $issuerDid;
        $this->signer = null;
        $this->configData = $configData;
    }

    public function withSigner(Signer $signer): IssuerStatePublisher
    {
        $this->signer = $signer->toProto();
        return $this;
    }

    public function build(): IssuerStateReceipt
    {
        $bridge = new Bridge();

        $req = new PublishIssuerStateRequest();
        $req->setConfigData($this->configData);
        $req->setIssuerDid($this->issuerDid);
        $req->setSigner($this->signer);

        $res = $bridge->identityV2->PublishIssuerState($req);

        if ($res->getError() != null) {
            throw new Exception($res->getError()->getMessage());
        }

        return IssuerStateReceipt::fromProto($res);
    }
}
